plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py

```diff
@@ -63,26 +66,34 @@ class SlurmSubmititConf(BaseSubmititConf):
 
 
 @dataclass
-class LocalSubmititConf(BaseSubmititConf):
-    pass
+class LocalParams(BaseParams):
+    executor: ExecutorName = ExecutorName.local  # can we get rid of this?
 
 
 @dataclass
-class SubmititConf:
-    executor: ExecutorName = ExecutorName.local
-    folder: str = "${hydra.sweep.dir}/.${hydra.launcher.params.executor}"
-    params: LocalSubmititConf = LocalSubmititConf()
+class SlurmConf(ObjectConf):
+    cls: str = "hydra_plugins.hydra_submitit_launcher.submitit_launcher.SubmititLauncher"
+    params: SlurmParams = SlurmParams()
 
 
 @dataclass
-class SubmititLauncherConf(ObjectConf):
+class LocalConf(ObjectConf):
     cls: str = "hydra_plugins.hydra_submitit_launcher.submitit_launcher.SubmititLauncher"
```

Comment:
You can define a launcher class for each mode and have and have the custom logic in it.
They can share a superclass or otherwise reuse most of the logic through other means.
```python
  cls: str = "hydra_plugins.hydra_submitit_launcher.submitit_launcher.SubmititLocalLauncher"
```
This will also allow you to named parameters in the `__init__` of the class:
```
class SubmititLauncher(Launcher):
    def __init__(self, 
      partition: Optional[str],
      comment: Optional[str],
      constraint: Optional[str],
      exclude: Optional[str],
      ...
) -> None:
        # forward directly to the implementation
        # (my preference would be to SlurmExecutor and LocalExecutor at this point).
        ...
```

Reply:
I'd rather not double down the name of each parameter both in the config and in the init. Naming them here would just add code for nothing in my opinion (we don't need type hints here since it's never called directly by a user, and all the checks can be efficiently performed through the config). So is there a shortcut to initialize the omega conf from a dictionary? I couldnt find any without requiring a for loop

Reply:
You asked how to simplify. this is my advice.
I don't think you will need to double much:
between config inheritance, and passing the actually needed parameters into each init there will be no duplication in my opinion.

regardless, it's your call.
Changing submitit can also help.

Reply:
but is there a better way to build the OmegaConf structure from a dict. Currently I am doing:
```python
        self.params = OmegaConf.structured(LocalParams)
        for key, val in params.items():
            OmegaConf.update(self.params, key, val)
```
bur I would think I am missing something (and it would not work with Enum, but fortunately there is no more Enum now).
Otherwise I'll use the Config directly but I prefer the flexibility of OmegaConf

Reply:
```
self.params = OmegaConf.structured(LocalParams(**params))
```

Keep in mind that dataclasses have a problem that causes them to not do what you expect if you instantiate:

```python
from dataclasses import dataclass

@dataclass
class A:
  a : int = 10

@dataclass
class B(A):
  a: int = 20

@dataclass
class C(A):
  a = 30

print(A())
print(B())
print(C())
```

```
$ python 1.py 
A(a=10)
B(a=20)
C(a=10)
```

If this is causing issues we can think of something.

Reply:
The general answer to:
> but is there a better way to build the OmegaConf structure from a dict

is:
```python
OmegaConf.create({})
```

In this example you want to get the type safety from the dataclass.
Here is some context:
https://omegaconf.readthedocs.io/en/latest/structured_config.html#merging-with-other-configs

You can also try what I suggested, which is possibly a better match here.